core/lib/spree/testing_support/factories/taxon_factory.rb

```diff
@@ -9,12 +9,19 @@
 
 FactoryBot.define do
   factory :taxon, class: 'Spree::Taxon' do
+    transient do
+      parent { nil }
+      taxonomy { parent&.taxonomy || create(:taxonomy) }
```

Comment:
Why does it need to be within the `transient` block? 🤔

Reply:
Interesting, I thought it was needed because `parent` and `taxonomy` are not attributes of Taxon. But the following works if you prefer it. I guess it works because `parent` and `taxonomy` are associations!
```ruby
  factory :taxon, class: 'Spree::Taxon' do
    name { 'Ruby on Rails' }
    taxonomy_id { (parent&.taxonomy || create(:taxonomy)).id }
    parent_id { parent&.id || taxonomy.root.id }

    trait :with_icon do
      after(:create) do |taxon|
        taxon.update(icon: Spree::Core::Engine.root.join('lib', 'spree', 'testing_support', 'fixtures', 'blank.jpg').open)
      end
    end
  end
```

Reply:
Yes, you can refer to the associations from the regular context 🙂

Reply:
Done, thanks!